### PR TITLE
add parameters the usepost mutation

### PR DESCRIPTION
**Pull request by Pierregvx**

### CODE REVIEW

File: web/massastation/src/custom/api/usePost.ts [hunk 1]
```diff
@@ -8,15 +8,28 @@ import { UseMutationResult, useMutation } from '@tanstack/react-query';
 
 export function usePost<T, P = unknown>(
   resource: string,
-): UseMutationResult<P, unknown, T, unknown> {
-  var url = `${import.meta.env.VITE_BASE_API}/${resource}`;
+): UseMutationResult<
+  P,
+  unknown,
+  { params?: Record<string, string>; payload?: T },
```
Comment (Thykof) [inline web/massastation/src/custom/api/usePost.ts:14]: can you explain this?

**Review by Thykof — CHANGES_REQUESTED** [review]
why don't you simply concatenate the query params in the ressource argument before calling the mutate?

File: web/massastation/src/custom/api/usePost.ts (hunk 1 above)
Comment (Pierregvx) [reply web/massastation/src/custom/api/usePost.ts:14]: these are the arguments that you give when we call the mutation,
ex
```ts
    const payload = { name: 'John Doe', age: 25 };
    const params = { token: 'abc123' };

    mutate({  payload, params });
```

**Review by mazmassa — APPROVED** [review]
We need to revisit that to not have a custom usePost for massastation. But we'll do in future.

**Comment by mazmassa** [comment]
> why don't you simply concatenate the query params in the ressource argument before calling the mutate?

Yep! I thought the same. But he needs a refreshed mutation. I mean, a mutation already with the correct url is filled on the field.

For now we have a custom format of usePost in massastation. (Weird!!!). But we have no time to do it differently.

**Pull request merged**